src/methods.jl

```diff
@@ -77,3 +77,59 @@ julia> subtract_overscan(frame, (:, 4:5), dims = 2)
 * [`subtract_overscan!`](@ref)
 """
 subtract_overscan(frame::AbstractArray, idxs; dims = axes_min_length(idxs)) = subtract_overscan!(deepcopy(frame), idxs, dims = dims)
+
+
+"""
+    flat_correct!(frame::AbstractArray, flat_frame::AbstractArray; min_value = nothing, norm_value = nothing)
+
+In place version of [`flat_correct`](@ref)
+"""
+function flat_correct!(frame::AbstractArray, flat_frame::AbstractArray; min_value = nothing, norm_value = nothing)
+    size(frame) != size(flat_frame) && error("size of frame and flat frame are not same")
+    !(norm_value === nothing) && norm_value < 0 && error("norm_value must be greater than 0")
+
+    !(min_value === nothing) && @. flat_frame[flat_frame < min_value] = min_value
+
+    if !(norm_value === nothing)
+        flat_frame ./= norm_value
+    else
+        flat_frame ./= mean(flat_frame)
+    end
+
+    frame ./= flat_frame
+    return frame
+end
+
+
+"""
+    flat_correct(frame::AbstractArray, flat_frame::AbstractArray; min_value = nothing, norm_value = nothing)
+
+Correct the image for flat fielding.
```

Comment:
This could be improved. I would write a short description like “Correct `frame` for non-conformity using the calibrated `flat_frame`” then followed by a more description of the process and the keyword arguments. This would be a good place to note the behavior of dividing by zero (do we error, do we mask it, do we return infinity).